
## Genome structure is not random

Why should we care about genome structure? Here we can see some microscopy
images from cell nuclei, using some stains to higlight diferent components. Top
left shows heterochromatin (DNA not used) being expelled to the perifery. Top
right depicts the transcriotion machinery in red and green. This is where action
happens: DNA information gets transcribed and eventually will give rise to
functional elements (proteins). There is a clear seggregation between active and
inactive chromatin, with spatial patterns associated with transcription. Bottom
left there are two different chromosomes tainted red and green. It becomes
apparent that chromosomes occopy defined regions inside the nucleus (chromosome
territories). We see two copies of each chromosome because this cell is diploid
for them. Bottom rigths shows again a chromosome 2 and a specific loci inside
it.

So, the spatial organization of DNA / chromatin is not random

## Chromosome conformation capture

It nice that we can see different patterns of DNA inside the nucleus, but
microscopy has limits in throughput and resolution. Biochemistry and molecular
biology comes to the rescue. In 2002 Dekker and co. presented a molecular assay
that provides information about relative spatial location of DNA. Basically they
make use uf fromaldehide that is a compound able to crosslink proteins present
in DNA. This way, things that are are in close proximity in the space become
bonded together. Then they used a restriction enzyme that is able to precisely
cut DNA following a known sequence pattern. After that, they put a mark on the
DNA ends and religate them. Finally, taking advantage of the marks, they perform
a pull down to capture the relevant DNA. At the end of the day, they obtain a
buch of quimeric reads providins information about spatial proximity of DNA
fragments. In 2009 Lieberman-Aiden extended the protocol to use it genome wide.
