<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Me &amp; myself</title>
    <meta charset="utf-8" />
    <meta name="author" content="Enrique Vidal @qenvio github.com/qenvio qenvio@gmail.com" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link href="libs/remark-css/hygge.css" rel="stylesheet" />
    <link href="libs/font-awesome/css/all.css" rel="stylesheet" />
    <link href="libs/font-awesome/css/v4-shims.css" rel="stylesheet" />
    <link rel="stylesheet" href="hola.css" type="text/css" />
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">







count: false
class: inverse, center, middle

&lt;/br&gt;
&lt;/br&gt;
&lt;/br&gt;

# .titan[Me + my stuff]

## 28 / 01 / 2020

&lt;/br&gt;

### Enrique (Quique) Vidal&lt;/br&gt;&lt;i class="fa fa-envelope"&gt;&lt;/i&gt; qenvio@gmail.com &lt;i class="fa fa-twitter"&gt;&lt;/i&gt; @qenvio &lt;i class="fa fa-github"&gt;&lt;/i&gt; qenvio


---
background-image: url(images/cabra.jpg)
background-size: contain

# Me


---

.center[
&lt;img src="images/uv.jpg" width="60%" /&gt;
]

--

.pull-left[
.center[
### Degree in physics
&lt;/br&gt;
&lt;img src="images/fisica.gif" width="80%" /&gt;
]
]

--

.pull-left[
.center[
### Degree in statistics
&lt;/br&gt;
&lt;img src="images/mates.gif" width="80%" /&gt;
]
]


---

.center[
&lt;img src="images/1_timeline.png" width="100%" /&gt;
]

.pull-left[
# Environmental and Cancer Epidemiology

]

---
count: false

.center[
&lt;img src="images/1_timeline.png" width="100%" /&gt;
]

.pull-left[
# Environmental and Cancer Epidemiology

### Observational studies

]


---
count: false

.center[
&lt;img src="images/1_timeline.png" width="100%" /&gt;
]

.pull-left[
# Environmental and Cancer Epidemiology

### Observational studies

### &amp;emsp; &lt;i class="fa fa-chart-line"&gt;&lt;/i&gt; Mortality trends


]

.pull-right[
.center[
&lt;/br&gt;
&lt;/br&gt;
&lt;/br&gt;
&lt;img src="images/changepoint.jpeg" width="100%" /&gt;
]
]

---
count: false

.center[
&lt;img src="images/1_timeline.png" width="100%" /&gt;
]

.pull-left[
# Environmental and Cancer Epidemiology

### Observational studies

### .make-transparent[&amp;emsp; &lt;i class="fa fa-chart-line"&gt;&lt;/i&gt; Mortality trends]

### &amp;emsp; &lt;i class="fa fa-industry"&gt;&lt;/i&gt; Cancer &amp; pollution

]



.pull-right[
.center[
&lt;/br&gt;
&lt;/br&gt;
&lt;img src="images/nuclear.png" width="100%" /&gt;
]
]

---
count: false

.center[
&lt;img src="images/1_timeline.png" width="100%" /&gt;
]

.pull-left[
# Environmental and Cancer Epidemiology

### Observational studies

### .make-transparent[&amp;emsp; &lt;i class="fa fa-chart-line"&gt;&lt;/i&gt; Mortality trends]

### &amp;emsp; &lt;i class="fa fa-industry"&gt;&lt;/i&gt; Cancer &amp; pollution

]



.pull-right[
.center[
&lt;img src="images/tarrega.png" width="80%" /&gt;
]
]

---
count: false

.center[
&lt;img src="images/1_timeline.png" width="100%" /&gt;
]

.pull-left[
# Environmental and Cancer Epidemiology

### .make-transparent[Observational studies]

### .make-transparent[&amp;emsp; &lt;i class="fa fa-chart-line"&gt;&lt;/i&gt; Mortality trends]

### .make-transparent[&amp;emsp; &lt;i class="fa fa-industry"&gt;&lt;/i&gt; Cancer &amp; pollution]

### &lt;i class="fa fa-user-graduate"&gt;&lt;/i&gt; PhD Epidemiology

]

.pull-right[
.center[
&lt;img src="images/tarrega.png" width="80%" /&gt;
&lt;img src="images/autonoma.gif" width="70%" /&gt;
]
]

---

&lt;img src="images/2_timeline.png" width="100%" /&gt;


.pull-left[
# Bioinformatics and Genomics

]

.pull-right[
]


---
count: false

&lt;img src="images/2_timeline.png" width="100%" /&gt;


.pull-left[
# Bioinformatics and Genomics

### &amp;emsp; &lt;i class="fa fa-project-diagram"&gt;&lt;/i&gt; Complex data
]

.pull-right[
]

---
count: false

&lt;img src="images/2_timeline.png" width="100%" /&gt;


.pull-left[
# Bioinformatics and Genomics

### &amp;emsp; &lt;i class="fa fa-project-diagram"&gt;&lt;/i&gt; Complex data
### &amp;emsp; &lt;i class="fa fa-microchip"&gt;&lt;/i&gt; HPC

]

.pull-right[
]


---
count: false

&lt;img src="images/2_timeline.png" width="100%" /&gt;


.pull-left[
# Bioinformatics and Genomics

### .make-transparent[&amp;emsp; &lt;i class="fa fa-project-diagram"&gt;&lt;/i&gt; Complex data]
### .make-transparent[&amp;emsp; &lt;i class="fa fa-microchip"&gt;&lt;/i&gt; HPC]

### Network analysis
]

.pull-right[
&lt;/br&gt;
&lt;/br&gt;
&lt;/br&gt;
&lt;/br&gt;
&lt;img src="images/network.jpeg" width="100%" /&gt;
]

---
count: false

&lt;img src="images/2_timeline.png" width="100%" /&gt;


.pull-left[
# Bioinformatics and Genomics

### .make-transparent[&amp;emsp; &lt;i class="fa fa-project-diagram"&gt;&lt;/i&gt; Complex data]
### .make-transparent[&amp;emsp; &lt;i class="fa fa-microchip"&gt;&lt;/i&gt; HPC]

### .make-transparent[Network analysis]
### Genetic diseases
]

.pull-right[
&lt;/br&gt;
&lt;/br&gt;
&lt;/br&gt;
&lt;/br&gt;
&lt;img src="images/pedigree.png" width="100%" /&gt;
]


---

&lt;img src="images/3_timeline.png" width="100%" /&gt;

.pull-left[
# Cancer Epigenetics and Biology

]

.pull-right[
]

---
count: false

&lt;img src="images/3_timeline.png" width="100%" /&gt;

.pull-left[
# Cancer Epigenetics and Biology

### &amp;emsp; &lt;i class="fa fa-database"&gt;&lt;/i&gt; Exploit available data
]

.pull-right[
]

---
count: false

&lt;img src="images/3_timeline.png" width="100%" /&gt;

.pull-left[
# Cancer Epigenetics and Biology

### &amp;emsp; &lt;i class="fa fa-database"&gt;&lt;/i&gt; Exploit available data
### &amp;emsp; &lt;i class="fa fa-user"&gt;&lt;/i&gt; One vs. all
]

.pull-right[
]

---
count: false

&lt;img src="images/3_timeline.png" width="100%" /&gt;

.pull-left[
# Cancer Epigenetics and Biology

### &amp;emsp; &lt;i class="fa fa-database"&gt;&lt;/i&gt; Exploit available data
### &amp;emsp; &lt;i class="fa fa-user"&gt;&lt;/i&gt; One vs. all
### &amp;emsp; &lt;i class="fa fa-thermometer-three-quarters"&gt;&lt;/i&gt; Under pressure
]

.pull-right[
]

---
count: false

&lt;img src="images/3_timeline.png" width="100%" /&gt;

.pull-left[
# Cancer Epigenetics and Biology

### &amp;emsp; .make-transparent[&lt;i class="fa fa-database"&gt;&lt;/i&gt; Exploit available data]
### &amp;emsp; .make-transparent[&lt;i class="fa fa-user"&gt;&lt;/i&gt; One vs. all]
### &amp;emsp; .make-transparent[&lt;i class="fa fa-thermometer-three-quarters"&gt;&lt;/i&gt; Under pressure]

### Data driven research
]

.pull-right[
&lt;/br&gt;
&lt;/br&gt;
&lt;img src="images/datadriven.png" width="100%" /&gt;
]

---

&lt;img src="images/4_timeline.png" width="100%" /&gt;

.pull-left[
# Chromatin and Gene Expression

]

.pull-right[
]

---
count: false

&lt;img src="images/4_timeline.png" width="100%" /&gt;

.pull-left[
# Chromatin and Gene Expression

### &amp;emsp; &lt;i class="fa fa-tasks"&gt;&lt;/i&gt; Lead a project

]

.pull-right[
]

---
count: false

&lt;img src="images/4_timeline.png" width="100%" /&gt;

.pull-left[
# Chromatin and Gene Expression

### &amp;emsp; &lt;i class="fa fa-tasks"&gt;&lt;/i&gt; Lead a project
### &amp;emsp; &lt;i class="fa fa-lightbulb"&gt;&lt;/i&gt; Find opportunities

]

.pull-right[
]

---
count: false

&lt;img src="images/4_timeline.png" width="100%" /&gt;

.pull-left[
# Chromatin and Gene Expression

### &amp;emsp; &lt;i class="fa fa-tasks"&gt;&lt;/i&gt; Lead a project
### &amp;emsp; &lt;i class="fa fa-lightbulb"&gt;&lt;/i&gt; Find opportunities
### &amp;emsp; &lt;i class="fa fa-handshake"&gt;&lt;/i&gt; Deal with stakeholders

]

.pull-right[
]

---
count: false

&lt;img src="images/4_timeline.png" width="100%" /&gt;

.pull-left[
# Chromatin and Gene Expression

### &amp;emsp; .make-transparent[&lt;i class="fa fa-tasks"&gt;&lt;/i&gt; Lead a project]
### &amp;emsp; .make-transparent[&lt;i class="fa fa-lightbulb"&gt;&lt;/i&gt; Find opportunities]
### &amp;emsp; .make-transparent[&lt;i class="fa fa-handshake"&gt;&lt;/i&gt; Deal with stakeholders]

### Genome structure

]

.pull-right[
&lt;/br&gt;
&lt;/br&gt;
&lt;/br&gt;
&lt;img src="images/3dgenome.jpg" width="100%" /&gt;
]



---


&lt;img src="images/0_timeline.png" width="100%" /&gt;

# What am I?


.center[
]

---
count:false 


&lt;img src="images/1_timeline.png" width="100%" /&gt;

# What am I?

## &lt;i class="fa fa-reddit-alien"&gt;&lt;/i&gt; Epidemiologist

.center[
]

---
count:false 


&lt;img src="images/23_timeline.png" width="100%" /&gt;

# What am I?

## .make-transparent[&lt;i class="fa fa-reddit-alien"&gt;&lt;/i&gt; Epidemiologist]

## &lt;i class="fa fa-reddit-alien"&gt;&lt;/i&gt; Bioinformatician


.center[
]

---
count:false 


&lt;img src="images/4_timeline.png" width="100%" /&gt;

# What am I?

## .make-transparent[&lt;i class="fa fa-reddit-alien"&gt;&lt;/i&gt; Epidemiologist]

## .make-transparent[&lt;i class="fa fa-reddit-alien"&gt;&lt;/i&gt; Bioinformatician]

## &lt;i class="fa fa-reddit-alien"&gt;&lt;/i&gt; Computational biologist

.center[
]

---
count:false 


&lt;img src="images/full_timeline.png" width="100%" /&gt;

# What am I?

## .make-transparent[&lt;i class="fa fa-reddit-alien"&gt;&lt;/i&gt; Epidemiologist]

## .make-transparent[&lt;i class="fa fa-reddit-alien"&gt;&lt;/i&gt; Bioinformatician]

## .make-transparent[&lt;i class="fa fa-reddit-alien"&gt;&lt;/i&gt; Computational biologist]

.center[
# Data scientist!
]


---

# Scientific production

.center[
&lt;img src="images/cites.png" width="70%" /&gt;
]

---
class: inverse

# My failures

--

## &amp;emsp; &lt;i class="fa fa-times-circle"&gt;&lt;/i&gt; PhD unpublished

--

### &amp;emsp; &amp;emsp; &amp;emsp;  &lt;i class="fa fa-hand-point-right"&gt;&lt;/i&gt; Wrap up and make public


--

## &amp;emsp; &lt;i class="fa fa-times-circle"&gt;&lt;/i&gt; Stakeholder high expectations

--

### &amp;emsp; &amp;emsp; &amp;emsp;  &lt;i class="fa fa-hand-point-right"&gt;&lt;/i&gt; Improve communication

--

## &amp;emsp; &lt;i class="fa fa-times-circle"&gt;&lt;/i&gt; Files and versions nightmare

--

### &amp;emsp; &amp;emsp; &amp;emsp;  &lt;i class="fa fa-hand-point-right"&gt;&lt;/i&gt; Automatization and version control

---

# Consulting for pharma industry

&lt;/br&gt;

--

## &amp;emsp; Carcinoma of unknown origin

--

## &amp;emsp; Epigenetic profiling

--

## &amp;emsp; Machine learning based diagnostic tool


---
count: false

# Consulting for pharma industry ...

&lt;/br&gt;


## &amp;emsp; .make-transparent[Carcinoma of unknown origin]


## &amp;emsp; .make-transparent[Epigenetic profiling]


## &amp;emsp; .make-transparent[Machine learning based diagnostic tool]

&lt;/br&gt;

.right[# ... apply my skills to make impact! &lt;i class="fa fa-smile-beam"&gt;&lt;/i&gt;]





---

# 4DGenome

.pull-left[
.center[
.theframe[
&lt;img src="images/M_Beato.jpg" width="90%" /&gt;
]
]
.center[
.theframe[
&lt;img src="images/T_Graff.jpg" width="90%" /&gt;
]
]
]

.pull-right[
.center[
.theframe[
&lt;img src="images/M_Marti.jpg" width="90%" /&gt;
]
]
.center[
.theframe[
&lt;img src="images/G_Filion.jpg" width="90%" /&gt;
]
]
]

---
count: false
class: inverse, center, middle


# .gozilla[whatever-C]


---

# Chromosome conformation capture

**Dekker, J. et al. (2002)**&lt;/br&gt;
*Capturing chromosome conformation*&lt;/br&gt;
Science

.center[
&lt;img src="images/crosslink.jpg" width="90%" /&gt;
]


**Lieberman-Aiden, E. et al. (2009)**&lt;/br&gt;
*Comprehensive mapping of long-range interactions reveals folding principles of
the human genome*&lt;/br&gt;
Science


---

# Hi-C Matrix: Step by step

.center[
&lt;img src="images/hic_animation_full.gif" width="60%" /&gt;
]

---

# Territories, compartments, domains, loops

.center[
&lt;img src="images/hic.png" width="70%" /&gt;
]


---

# Capture-C: Contact matrices

.center[
&lt;img src="images/chic_1.png" width="75%" /&gt;
]

---
count: false

# Capture-C: Contact matrices

.center[
&lt;img src="images/chic_2.png" width="75%" /&gt;
]

---

# Capture-C: Contact profile

.center[
&lt;img src="images/virtual_1.png" width="75%" /&gt;
]

---
count: false

# Capture-C: Contact profile

.center[
&lt;img src="images/virtual_2.png" width="75%" /&gt;
]


---

# 4C

.center[
&lt;img src="images/profile_domainogram.png" width="70%" /&gt;
]




---
count: false
class: inverse, center, middle

# .gozilla[Data production]

---

# Samples produced and processed

.center[
&lt;img src="images/processed_samples.png" width="75%" /&gt;
]


---
count: false
class: inverse, center, middle

# .gozilla[Challenges]

---
count: false
class: inverse

# Challenges

--

## &amp;emsp; 1. Sequencing overflow

---

# Sequencing overflow

### Over 1,000 samples processed

--
## &amp;emsp; Metadata

--
## &amp;emsp; Automatization

--
## &amp;emsp; Cluster (storage and jobs)

--
## &amp;emsp; Exploration and visualization aids

---

# Parallel sequencing lives

.center[
.theframe[
&lt;img src="images/parallel_1.png" width="100%" /&gt;
]
]

---

# Parallel sequencing lives


.center[
.theframe[
&lt;img src="images/parallel_2.jpeg" width="90%" /&gt;
]
]


---

# HIsee: Metadata

.center[
.theframe[
&lt;img src="images/metadata.gif" width="90%" /&gt;
]
]

---

# HIsee: Quality control

.center[
.theframe[
&lt;img src="images/qc.gif" width="90%" /&gt;
]
]

---

# HIsee: Contact matrices

.center[
.theframe[
&lt;img src="images/contacts.gif" width="90%" /&gt;
]
]

---

# HIsee: Whole chromosome

.center[
.theframe[
&lt;img src="images/contacts_wholechr.gif" width="90%" /&gt;
]
]

---

# HIsee: Smoothing!

.center[
.theframe[
&lt;img src="images/contacts_detail.gif" width="90%" /&gt;
]
]

---

# Virual 4C

.center[
.theframe[
&lt;img src="images/virtual4c.gif" width="90%" /&gt;
]
]


---
count: false
class: inverse

# Challenges

## &amp;emsp; 1. Sequencing overflow

## &amp;emsp; 2. Reproducibility

---

# Reproducibility

--

## &amp;emsp; Biases

--

## &amp;emsp; Aberrant karyotypes

--

## &amp;emsp; Incomplete designs

--

## &amp;emsp; Computational resources



---

# Biases

.center[
&lt;img src="images/toy_signal.png" width="70%" /&gt;
]

---
count: false

# Biases

.center[
&lt;img src="images/toy_signal_and_bias.png" width="70%" /&gt;
]

---
count: false

# Biases

.center[
&lt;img src="images/toy_signal_and_bias_and_decay.png" width="70%" /&gt;
]


---
count: false

# Biases

.center[
&lt;img src="images/toy_simulated.png" width="70%" /&gt;
]

---

# Aberrant karyotypes

.center[
&lt;img src="images/hic_copy-number_only-raw.png" width="70%" /&gt;
]

---

# Inclomplete designs

.center[
&lt;img src="images/capture.png" width="70%" /&gt;
]




---

# A model-based approach to bias removal

.center[
&lt;img src="images/hic_totals_filtered.png" width="60%" /&gt;
]

---
count: false

# A model-based approach to bias removal

.center[
&lt;img src="images/hic_totals_smooth_filtered.png" width="60%" /&gt;
]

---

# A model-based approach to bias removal

&lt;br&gt;

.center[
.theframe[
&lt;img src="images/oned_1.png" width="90%" /&gt;
]
]

&lt;/br&gt;
.center[
### &lt;i class="fa fa-github"&gt;&lt;/i&gt; qenvio/dryhic
]

---

# OneD vs. the world

.center[
&lt;img src="images/benchmark_values_rocs_all.png" width="70%" /&gt;
]


---

# OneD is thrifty

--

.pull-left[
.center[
&lt;img src="images/hic_totals_subsample_correlation_filtered.png" width="100%" /&gt;
]
]

--

.pull-right[
.center[
&lt;img src="images/hic_resolution_projection_filtered.png" width="100%" /&gt;
]
]


---

# OneD is fast

.center[
&lt;img src="images/reso_times.png" width="70%" /&gt;
]


---
count: false
class: inverse

# Challenges

## &amp;emsp; 1. Sequencing overflow

## &amp;emsp; 2. Reproducibility

## &amp;emsp; 3. Data &amp;#8594; Info &amp;#8594; Knowledge

---

# The inner life of TADs

--
.pull-left[
.center[
&lt;img src="images/intratad_cartoon.png" width="100%" /&gt;
]
]

--
.pull-right[
.center[
&lt;img src="images/intratad_raw_onebatch.png" width="100%" /&gt;
]
]

---

# The inner life of TADs

.pull-left[
.center[
&lt;img src="images/intratad_cartoon.png" width="100%" /&gt;
]
]

.pull-right[
.center[
&lt;img src="images/intratad_linearmodel_onebatch.png" width="100%" /&gt;
]
]

---

# The inner life of TADs

.pull-left[
.center[
&lt;img src="images/intratad_cartoon.png" width="100%" /&gt;
]
]

.pull-right[
.center[
&lt;img src="images/intratad_linearmodel_foldchange_onebatch.png" width="100%" /&gt;
]
]


---

# Hormone-control regions

.center[
.theframe[
&lt;img src="images/hormone_2.png" width="100%" /&gt;
]
]

---

# Hormone-control regions

.center[
.theframe[
&lt;img src="images/hormone_1.png" width="70%" /&gt;
]
]

---

# 3D genome and gene regulation

.center[
.theframe[
&lt;img src="images/tf_2.png" width="70%" /&gt;
]
]

---

# 3D genome and gene regulation

.center[
.theframe[
&lt;img src="images/tf_1.png" width="100%" /&gt;
]
]



---
count: false
class: inverse

# Challenges

## &amp;emsp; 1. Sequencing overflow

## &amp;emsp; 2. Reproducibility

## &amp;emsp; 3. Data &amp;#8594; Info &amp;#8594; Knowledge

## &amp;emsp; 4. Be in the middle


---

# Collaborations and scientific production


.center[
&lt;img src="images/papers.png" width="70%" /&gt;
]



---
count: false
class: inverse, center, middle

# .gozilla[To-Do]

---

# Consensus

--

## &amp;emsp; Quality control

--

## &amp;emsp; Data formats and storage

--

## &amp;emsp; Callers (A/B, TAD, loop ...)


---

# So ...

&lt;br&gt;

&lt;center&gt;
&lt;blockquote class="twitter-tweet"&gt;&lt;p lang="en" dir="ltr"&gt;A DNA sequencer is a machine that automates the generation of batch effects.&lt;/p&gt;&amp;mdash; Lior Pachter (@lpachter) &lt;a href="https://twitter.com/lpachter/status/1161934792335560704?ref_src=twsrc%5Etfw"&gt;August 15, 2019&lt;/a&gt;&lt;/blockquote&gt; &lt;script async src="https://platform.twitter.com/widgets.js" charset="utf-8"&gt;&lt;/script&gt;
&lt;/center&gt;
&lt;script async src="https://platform.twitter.com/widgets.js" charset="utf-8"&gt;&lt;/script&gt;


---
count: false 

# So ... do we have batch effect? ...

&lt;br&gt;

&lt;center&gt;
&lt;blockquote class="twitter-tweet"&gt;&lt;p lang="en" dir="ltr"&gt;A DNA sequencer is a machine that automates the generation of batch effects.&lt;/p&gt;&amp;mdash; Lior Pachter (@lpachter) &lt;a href="https://twitter.com/lpachter/status/1161934792335560704?ref_src=twsrc%5Etfw"&gt;August 15, 2019&lt;/a&gt;&lt;/blockquote&gt; &lt;script async src="https://platform.twitter.com/widgets.js" charset="utf-8"&gt;&lt;/script&gt;
&lt;/center&gt;
&lt;script async src="https://platform.twitter.com/widgets.js" charset="utf-8"&gt;&lt;/script&gt;


---

# So ... do we have batch effect? ... Of course


.center[
&lt;img src="images/intratad_batch_pca_raw.png" width="60%" /&gt;
]

---
count: false

# OK ... Can we fix it?

.center[
&lt;img src="images/intratad_batch_pca_raw.png" width="60%" /&gt;
]

---

# OK ... Can we fix it? ... It depends


.center[
&lt;img src="images/intratad_batch_pca_batch.png" width="60%" /&gt;
]

---

# Data release

--

&lt;br&gt;

## Do you remember we have generated +500 samples?

--

&lt;br&gt;

## Well ... now we have to make them public



---
class: inverse, center



# .titan[4DGenome unit]

--

&lt;/br&gt;
 

# It's about C technologies


--

&lt;/br&gt;


# Ask us

--

 
## (Before we leave)


---
count: false

# ROC and AUC normal karyotypes

.center[
&lt;img src="images/benchmark_values_rocs_normals.png" width="70%" /&gt;
]

---
count: false

# PRC and AUC

.center[
&lt;img src="images/benchmark_values_prcs_all.png" width="70%" /&gt;
]
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="libs/remark-latest.min.js"></script>
<script src="https://platform.twitter.com/widgets.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false,
"self_contained": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
